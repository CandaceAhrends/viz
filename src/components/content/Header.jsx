import React, { useContext } from 'react';
import SideMenu from '../navigation/SideMenu';
import { StockContext } from '../../StockContext';
import HamIconToggle from './HamIconToggle';
import { useAppDispatch, useAppSelector } from '../../hooks';
import './content.scss';

const Header = () => {
  const { selectedStock, selectedQuote } = useContext(StockContext);

  const selectedMenu = useAppSelector((state) => state.navigation.selectedMenu);

  return (
    <header className="text-white h-[40px]  mt-1  border-b border-neutral-900 ">
      <div className="flex justify-center text-green ">
        {selectedMenu === 'charts' ? (
          <p>Active Charts</p>
        ) : (
          <div className="stock-display text-sm">
            <span className="stock-symbol">{selectedStock}</span>

            <span className="stock-change">
              {Number(selectedQuote?.todaysChange).toFixed(2)}
            </span>
            <span className="stock-change">
              {Number(selectedQuote?.todaysChangePerc).toFixed(2)}%
            </span>
          </div>
        )}
      </div>
      <div className="flex justify-end w-full p-1 mr-5 mt-[-1.2rem] z-[999999]">
        <HamIconToggle Menu={SideMenu} />
      </div>
    </header>
  );
};

export default Header;
