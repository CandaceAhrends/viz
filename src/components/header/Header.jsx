import React, { useContext } from 'react';
import SideMenu from '../navigation/SideMenu';
import HamIconToggle from '../navigation/HamIconToggle';
import { useAppDispatch, useAppSelector } from '../../hooks';
import dayjs from 'dayjs';
import './header.scss';

const Header = () => {
  const selectedStockDate = dayjs(useAppSelector((state) => state.stocks.date));
  const selectedStock = useAppSelector(
    (state) => state.historicalData.selectedStock
  );
  const selectedMenu = useAppSelector((state) => state.navigation.selectedMenu);

  return (
    <header className="text-white h-[40px]  mt-1  border-b border-neutral-900 ">
      <div className="flex justify-center text-green ">
        <div className="stock-display text-sm">
          <span className="stock-symbol">{selectedStock?.symbol}</span>

          <span className="stock-change">
            {selectedStockDate.format('dddd, MMMM D, YYYY')}
          </span>
        </div>
      </div>
      <div className="flex justify-end w-full p-1 mr-5 mt-[-1.2rem] z-[999999]">
        <HamIconToggle Menu={SideMenu} />
      </div>
    </header>
  );
};

export default Header;
