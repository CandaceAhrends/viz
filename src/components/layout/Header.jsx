import React from 'react';
import SideMenu from '../navigation/SideMenu.jsx';
import HamIconToggle from '../navigation/HamIconToggle.jsx';
import { useAppSelector } from '../../hooks';
import dayjs from 'dayjs';
import GitHubSvg from '../images/GitHubSvg.jsx';

const Header = () => {
  const selectedStockDate = dayjs(useAppSelector((state) => state.stocks.date));
  const selectedStock = useAppSelector(
    (state) => state.historicalData.selectedStock
  );

  return (
    <header className="text-white h-[40px]  mt-1  border-b border-neutral-900 ">
      <a
        target="_blank"
        href="https://github.com/CandaceAhrends/viz"
        className="flex items-center  text-white text-xs  p-2 absolute top-[-.5rem] left-3 md:left-20"
      >
        <GitHubSvg />
        <span className="m-2 mt-5">Code</span>
      </a>
      <div className="flex justify-center text-green ">
        <div className="stock-display text-sm">
          <span className="stock-symbol">{selectedStock?.symbol}</span>

          <span className="stock-date">
            {selectedStockDate.format('dddd, MMMM D, YYYY')}
          </span>
        </div>
      </div>
      <div className="flex justify-end w-full p-1 mr-5 mt-[-1.2rem] z-[999999]">
        <HamIconToggle Menu={SideMenu} />
      </div>
    </header>
  );
};

export default Header;
