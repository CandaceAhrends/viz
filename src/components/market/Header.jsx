import React, { useEffect } from 'react';
import { useAppDispatch, useAppSelector } from '../../hooks';
import StockSummary from '../ticker/StockSummary';
import './market.scss';

const Header = () => {
  const [summaryStocks, setSummaryStocks] = React.useState([]);
  const selectedStock = useAppSelector(
    (state) => state.historicalData.selectedStock
  );
  const marketSummary = useAppSelector(
    (state) => state.historicalData.marketSummary
  );
  useEffect(() => {
    if (marketSummary) {
      setSummaryStocks([...marketSummary, selectedStock]);
    }
  }, [selectedStock]);

  return (
    <header className="w-[27rem] md:w-[30rem] m-auto">
      <div className="header">
        <h1>
          Stockmarket<span>viz</span>
        </h1>
        <div className="flex justify-center text-orange-400">
          <span className="mr-1">!</span>
          <h5>Current market data is delayed by at least 15 minutes</h5>
          <span className="ml-1">:(</span>
        </div>
      </div>

      <div className="flex justify-center">
        {summaryStocks?.length > 1 && (
          <StockSummary summaryStocks={summaryStocks} />
        )}
      </div>
    </header>
  );
};

export default Header;
