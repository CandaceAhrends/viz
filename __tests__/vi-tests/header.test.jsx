import React from 'react';
import { renderWithProviders } from '../mock-store';
import { describe, it, expect, vi } from 'vitest';
import Header from '../../src/components/header/Header';
import dayjs from 'dayjs';

const preloadedState = {
  stocks: {
    date: dayjs(),
  },
  historicalData: {
    selectedStock: {
      symbol: 'MOCK',
    },
    filteredStocks: [],
  },
};

describe('Header', () => {
  it('renders without error', () => {
    const { container } = renderWithProviders(<Header />, { preloadedState });
    const currentDate = dayjs().format('dddd, MMMM D, YYYY');
    const element = container.getElementsByClassName('stock-date');
    expect(element[0].textContent).toEqual(currentDate);
  });
});
