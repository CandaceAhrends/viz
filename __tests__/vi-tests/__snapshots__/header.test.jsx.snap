// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders without error 1`] = `"<header class="text-white h-[40px]  mt-1  border-b border-neutral-900 "><a target="_blank" href="https://github.com/CandaceAhrends/viz" class="flex items-center  text-white text-xs  p-2 absolute top-[-.5rem] left-3 md:left-20"><svg width="42" height="42" viewBox="0 0 98 96" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M48.854 0C21.839 0 0 22 0 49.217c0 21.756 13.993 40.172 33.405 46.69 2.427.49 3.316-1.059 3.316-2.362 0-1.141-.08-5.052-.08-9.127-13.59 2.934-16.42-5.867-16.42-5.867-2.184-5.704-5.42-7.17-5.42-7.17-4.448-3.015.324-3.015.324-3.015 4.934.326 7.523 5.052 7.523 5.052 4.367 7.496 11.404 5.378 14.235 4.074.404-3.178 1.699-5.378 3.074-6.6-10.839-1.141-22.243-5.378-22.243-24.283 0-5.378 1.94-9.778 5.014-13.2-.485-1.222-2.184-6.275.486-13.038 0 0 4.125-1.304 13.426 5.052a46.97 46.97 0 0 1 12.214-1.63c4.125 0 8.33.571 12.213 1.63 9.302-6.356 13.427-5.052 13.427-5.052 2.67 6.763.97 11.816.485 13.038 3.155 3.422 5.015 7.822 5.015 13.2 0 18.905-11.404 23.06-22.324 24.283 1.78 1.548 3.316 4.481 3.316 9.126 0 6.6-.08 11.897-.08 13.526 0 1.304.89 2.853 3.316 2.364 19.412-6.52 33.405-24.935 33.405-46.691C97.707 22 75.788 0 48.854 0z" fill="#fff"></path></svg><span class="m-2 mt-5">Code</span></a><div class="flex justify-center text-green "><div class="stock-display text-sm"><span class="stock-symbol">MOCK</span><span class="stock-change">Saturday, January 11, 2025</span></div></div><div class="flex justify-end w-full p-1 mr-5 mt-[-1.2rem] z-[999999]"><div class="mr-[1rem]"><ul class="group  flex flex-col items-center gap-1.5 cursor-pointer h-[2rem]"><li class="w-8 h-[.2rem] rounded-full bg-white transition group-[.is-open]:rotate-45 group-[.is-open]:translate-y-[.79rem]"></li><li class="w-8 h-[.2rem] rounded-full bg-white group-[.is-open]:scale-x-0 transition"></li><li class="w-8 h-[.2rem] rounded-full bg-white group-[.is-open]:-rotate-45 group-[.is-open]:-translate-y-[.35rem]"></li></ul><div class="sidebar-container z-[999999]"><div class="sidebar bg-gray-800 text-white h-full overflow-scroll closed transition-all duration-500"><ul class="flex flex-col pt-5  "></ul></div></div></div></div></header>"`;
